components/automate-ui/src/app/helpers/auth/id-mapper.spec.ts

```diff
@@ -4,11 +4,15 @@ import { IdMapper } from './id-mapper';
 describe('IdMapper', () => {
 
     using([
-      ['my-name', 'my-name', 'mirrors nominal input'],
-      ['-0a5l9z-', '-0a5l9z-', 'supports numbers, letters, and hyphens'],
-      ['my-name!', 'my-name-', 'replaces trailing out-of-range char with hyphen'],
-      ['#$my-name', '--my-name', 'replaces leading out-of-range chars with hyphens'],
-      ['my name', 'my-name', 'replaces space char with hyphen'],
+      ['-0a5l9z-', '-0a5l9z-', 'preserves numbers, lowercase letters, and hyphens'],
+      ['my_name__', 'my_name__', 'preserves underscores'],
```

Comment:
This is the new test case that started red and is now green.
(Other test cases were added/changed here to increase the depth of coverage.)